File: app/src/main/java/com/example/starwars/viewmodel/FilmActivityViewModel.java
package com.example.starwars.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.example.starwars.model.Film;
import com.example.starwars.repository.FilmsRepository;

public class FilmActivityViewModel extends ViewModel {

    private FilmsRepository mRepository;
    private LiveData<Film> mFilmLiveData;
    private LiveData<Boolean> mIndicator;

    public void init(int id) {
        mRepository = FilmsRepository.getInstance();
        mIndicator = mRepository.getLoadingIndicator();
        mFilmLiveData = mRepository.getFilmFromAPI(id);
    }

    public LiveData<Film> getFilmLiveData() {
        return mFilmLiveData;
    }

    public LiveData<Boolean> getIndicator() {
        return mIndicator;
    }
}

File: app/src/main/java/com/example/starwars/viewmodel/FilmActivityViewModel.kt
package com.example.starwars.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.example.starwars.model.Film
import com.example.starwars.repository.FilmsRepository
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class FilmActivityViewModel @Inject constructor(private var mRepository: FilmsRepository) : ViewModel() {
    lateinit var filmLiveData: LiveData<Film?>
        private set
    lateinit var indicator: LiveData<Boolean>
        private set
    private var mId = 0

    fun init(id: Int) {
        mId = id
        indicator = mRepository.loadingIndicator
    }

    fun loadData(isInternetConnected: Boolean) {
        if (isInternetConnected) filmLiveData = mRepository.getFilmFromAPI(mId)
        else filmLiveData = mRepository.getFilmFromDatabase(mId)
    }
}